From 4f0ce1f34c99685ab9f5cdeb621280e6c05f42a8 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Fri, 2 Oct 2015 17:31:33 +0200
Subject: [PATCH] configure: Make libssh2 usage optional and independent of
 Curl checks

Curl can optionally use libssh2, but there's a realistic chance that other
plugins might need/support this library in the future as well
(since SSH support is fairly widespread), so make it its own separate
check.

Upstream-Status: Submitted [https://bugzilla.gnome.org/show_bug.cgi?id=755989]

Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 configure.ac | 31 ++++++++++++++++++++++---------
 1 file changed, 22 insertions(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index 28a01e7..79b0e12 100644
--- a/configure.ac
+++ b/configure.ac
@@ -575,6 +575,28 @@ else
   AG_GST_DISABLE_PLUGIN(dccp)
 fi
 
+dnl *** libssh2 ***
+AC_ARG_ENABLE([libssh2],
+     [  --enable-libssh2        enable libssh2 support @<:@default=auto@:>@],
+     [case "${enableval}" in
+       yes)  NEED_SSH2=yes ;;
+       no)   NEED_SSH2=no ;;
+       auto) NEED_SSH2=auto ;;
+       *) AC_MSG_ERROR([bad value ${enableval} for --enable-libssh2]) ;;
+     esac],[NEED_SSH2=auto])
+
+if test "x$NEED_SSH2" != "xno"; then
+  PKG_CHECK_MODULES(SSH2, libssh2 >= 1.4.3, [
+    HAVE_SSH2="yes"
+    AC_DEFINE(HAVE_SSH2, 1, [Define if libssh2 is available])
+  ], [
+    HAVE_SSH2="no"
+  ])
+fi
+AM_CONDITIONAL(USE_SSH2, test "x$HAVE_SSH2" = "xyes")
+AC_SUBST(SSH2_CFLAGS)
+AC_SUBST(SSH2_LIBS)
+
 dnl *** opengl ***
 AC_ARG_ENABLE([opengl],
      [  --enable-opengl         Enable Desktop OpenGL support @<:@default=auto@:>@],
@@ -2068,15 +2090,6 @@ AG_GST_CHECK_FEATURE(CURL, [Curl plugin], curl, [
   ])
   AC_SUBST(CURL_CFLAGS)
   AC_SUBST(CURL_LIBS)
-  PKG_CHECK_MODULES(SSH2, libssh2 >= 1.4.3, [
-    HAVE_SSH2="yes"
-    AC_DEFINE(HAVE_SSH2, 1, [Define if libssh2 is available])
-  ], [
-    HAVE_SSH2="no"
-  ])
-  AM_CONDITIONAL(USE_SSH2, test "x$HAVE_SSH2" = "xyes")
-  AC_SUBST(SSH2_CFLAGS)
-  AC_SUBST(SSH2_LIBS)
 ],,,[AM_CONDITIONAL(USE_SSH2, false)])
 
 dnl **** DASH ****
-- 
1.9.1

